File: cpp/888_公平的糖果棒交换_easy.cpp
class Solution {
public:
    vector<int> fairCandySwap(vector<int>& A, vector<int>& B) {
        int totalA = accumulate(A.begin(), A.end(), 0);
        int totalB = accumulate(B.begin(), B.end(), 0);
        int target = (totalA+totalB)/2;
        vector <int> ans(2);
        unordered_set<int> setB(B.begin(), B.end());
        for (auto a : A) {
            int need = target - totalA + a;
            if (setB.count(need)) {
                ans = vector<int>{a, need};
                break;
            }
        }
        return ans;
    }
};

File: python/888_公平的糖果棒交换_easy.py
class Solution:
    def fairCandySwap(self, A: List[int], B: List[int]) -> List[int]:
        totalA = sum(A)
        totalB = sum(B)
        target = (totalA+totalB)//2
        setB = set(B)
        for a in A:
            need = target-totalA+a
            if need in B:
                return [a, need]
        return -1